installation_and_upgrade/ibex_install_utils/tasks/backup_tasks.py

```diff
@@ -79,19 +74,20 @@ def copy_function(src, dst):
 
             current_file_index += 1
             self.update_progress_bar(current_file_index, number_of_files)
-
-        shutil.copytree(src, dst, ignore=ignore, copy_function=count_files_copy_function)
+            
         shutil.copytree(src, dst, ignore=ignore, copy_function=copy_function, dirs_exist_ok=True)
     
 
-    def _check_backup_space(self, src):
+    def _check_backup_space(self, src, ignore=None):
         # Checks if there is enough space to move dir at src into the backup directory
         # (all in bytes)
         _, _, free = shutil.disk_usage(BACKUP_DIR)
-        backup_size = FileUtils.get_size(src)
+        backup_size, number_of_files = FileUtils.get_size_and_number_of_files(src, ignore=ignore)
         if backup_size > free:
             needed_space = round((backup_size - free) / (1024 ** 3), 2)
             self.prompt.prompt_and_raise_if_not_yes(f"You don't have enough space to backup. Free up {needed_space} GB at {BACKUP_DIR}")
+        else:
+            return backup_size, number_of_files
```

Comment:
does the return need to be outside rhe else: - what happens if the backup_size > free case does not throw an exceptiopn, or does it always?

Reply:
That's a good point yes, well spotted.